app/models/course/discussion/topic.rb

```diff
@@ -13,4 +13,12 @@ class Course::Discussion::Topic < ActiveRecord::Base
   def subscribed_by?(user)
     subscriptions.where(user: user).any?
   end
+
+  # Create subscription for a user
+  #
+  # @param [User] user The user who needs to subscribe to this topic
+  def ensure_subscribed_by(user)
+    return if self.subscribed_by?(user)
+    subscriptions.build(user: user).save!
```

Comment:
I've used `save!`, so that if save successfully it will return `true` otherwise it will raise exceptions. @lowjoel